common/src/main/java/com/wynntils/commands/LootrunCommand.java

```diff
@@ -0,0 +1,404 @@
+                                                                    "/lootrun save "))))
+                                            .withStyle(ChatFormatting.GREEN)),
+                            false);
+        }
+        return 1;
+    }
+
+    private int saveLootrun(CommandContext<CommandSourceStack> context) {
+        String name = StringArgumentType.getString(context, "name");
+        LootrunUtils.LootrunSaveResult lootrunSaveResult = LootrunUtils.saveLootrun(name);
+        switch (lootrunSaveResult) {
+            case SAVED -> {
+                context.getSource()
+                        .sendSuccess(
+                                new TranslatableComponent("feature.wynntils.lootrunUtils.savedLootrun")
+                                        .withStyle(ChatFormatting.GREEN),
+                                false);
+                return 1;
+            }
+            case ERROR_SAVING -> {
+                context.getSource()
+                        .sendFailure(new TranslatableComponent("feature.wynntils.lootrunUtils.errorSavingLootrun")
+                                .withStyle(ChatFormatting.RED));
+                return 0;
+            }
+            case ERROR_ALREADY_EXISTS -> {
+                context.getSource()
+                        .sendFailure(new TranslatableComponent(
+                                        "feature.wynntils.lootrunUtils.errorSavingLootrunAlreadyExists")
+                                .withStyle(ChatFormatting.RED));
+                return 0;
+            }
+        }
+        return 0;
+    }
+
+    private int addJsonLootrunNote(CommandContext<CommandSourceStack> context) {
+        Component text = ComponentArgument.getComponent(context, "text");
+        Entity root = McUtils.player().getRootVehicle();
+        BlockPos pos = root.blockPosition();
+        context.getSource()
+                .sendSuccess(
+                        new TranslatableComponent("feature.wynntils.lootrunUtils.addedNote", pos.toShortString())
+                                .append("\n" + text),
+                        false);
+        return LootrunUtils.addNote(text);
+    }
+
+    private int addTextLootrunNote(CommandContext<CommandSourceStack> context) {
+        Component text = new TextComponent(StringArgumentType.getString(context, "text"));
+        Entity root = McUtils.player().getRootVehicle();
+        BlockPos pos = root.blockPosition();
+        context.getSource()
+                .sendSuccess(
+                        new TranslatableComponent("feature.wynntils.lootrunUtils.addedNote", pos.toShortString())
+                                .append("\n" + text),
+                        false);
+        return LootrunUtils.addNote(text);
+    }
+
+    private int listLootrunNote(CommandContext<CommandSourceStack> context) {
+        LootrunUncompiled current = LootrunUtils.getActiveLootrun();
```

Comment:
it would make more sense to replace this with a `getNotes` function, no? with a similar sort of null check